docs/manual/commands/index.rst

```diff
@@ -1,309 +1,54 @@
-for Qtile.
+Qtile's command API is based on a graph of objects, where each object has a set
+of associated commands, combined with a number of interfaces that are used
+to navigate the graph and execute associated commands.
 
-The other primary example of this is the ``IPCCommandInterface`` which is able
-to then route all calls through an IPC client connected to a running Qtile
-instance.  In this case, the command graph call can be constructed on the
-client side without having to dispatch to Qtile and once the call is
-constructed and deemed valid, the call can be executed.
+This page gives an overview of the command graph and the various interfaces
+accessible by users. The documentation also contains details of all the commands
+that are exposed by objects on the graph.
 
-In both of these cases, executing a command on a command interface will return
-the result of executing the command on a running Qtile instance.  To support
-lazy execution, the ``LazyCommandInterface`` instead returns a ``LazyCall``
-which is able to be resolved later by the running Qtile instance when it is
-configured to fire.
+.. note::
 
-Tying it together: Command Client
-=================================
+  While users are able to access the internal python objects (e.g. via a ``qtile``
+  instance), this is not part of the "official" API. These objects and method are
+  not currently included in the documentation but can be viewed by looking at the
+  source code on github. Changes to commonly-used internal objects will be kept to
+  a minimum.
 
-So far, we have our running Command Objects and the Command Interface to
-dispatch commands against these objects as well as the Command Graph structure
-itself which encodes how to traverse the connections between the objects.  The
-final component which ties everything together is the Command Client, which
-allows us to navigate through the graph to resolve objects, find their
-associated commands, and execute the commands against the held command
-interface.
+The graph and object commands are used in a number of
+different places:
 
-The idea of the command client is that it is created with a reference into the
-command graph and a command interface.  All navigation can be done against the
-command graph, and traversal is done by creating a new command client starting
-from the new node.  When a command is executed against a node, that command is
-dispatched to the held command interface.  The key decision here is how to
-perform the traversal.  The command client exists in two different flavors: the
-standard ``CommandClient`` which is useful for handling more programatic
-traversal of the graph, calling methods to traverse the graph, and the
-``InteractiveCommandClient`` which behaves more like a standard Python object,
-traversing by accessing properties and performing key lookups.
+* Commands can be :ref:`bound to keys <config-keys>` in the Qtile
```

Comment:
Same here

Reply:
As above, these bullets build for me locally.